core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -332,7 +332,8 @@ private[spark] class Executor(
 
   class TaskRunner(
       execBackend: ExecutorBackend,
-      private val taskDescription: TaskDescription)
+      private val taskDescription: TaskDescription,
+      private val plugins: Option[PluginContainer])
```

Comment:
Maybe we can make the `plugins` parameter optional or default to some `EmptyPluginContainer`?:
```suggestion
      private val plugins: Option[PluginContainer] = None)
```

Reply:
Same for `Task#run`.

Reply:
@rshkv, what is the reason to make this default to None?  This is an internal api and only called from here. It's an option already so people can check it easily.  In some ways its nice to force it so you make sure all uses of it have been updated.  
Are there cases you know this is used outside Spark?